tests/Integration/Behaviour/Features/Context/CartRuleFeatureContext.php

```diff
@@ -156,30 +164,36 @@ protected function createCartRule(
         $cartRule->date_to = $now->format('Y-m-d H:i:s');
         $cartRule->active = true;
         $cartRule->add();
-        $this->cartRules[$cartRuleName] = $cartRule;
-        SharedStorage::getStorage()->set($cartRuleName, $cartRule->id);
+        $this->getSharedStorage()->set($cartRuleName, $cartRule->id);
     }
 
     /**
      * @Given /^cart rule "(.+?)" is restricted to the category "(.+?)" with a quantity of (\d+)$/
      */
     public function cartRuleWithProductRuleRestriction(string $cartRuleName, string $categoryName, int $quantity)
     {
-        $this->checkCartRuleWithNameExists($cartRuleName);
+        if ($this->getSharedStorage()->exists($cartRuleName)) {
+            //@todo: This allows applying this step to cart rule that was created with a step using CQRS command and saved to shared storage
+            //       it is not ideal, but for now it should work, until restrictions are migrated.
+            $cartRuleId = $this->getSharedStorage()->get($cartRuleName);
+        } else {
+            $this->checkCartRuleWithNameExists($cartRuleName);
+            $cartRuleId = $this->cartRules[$cartRuleName]->id;
+        }
```

Comment:
You use this piece of code at least four times, it's worth putting this in a method I think